# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentChannelParent.h
# Commit: 81ddcb21f057
# Full Hash: 81ddcb21f05754f2b3b4e581e5d4295fada1cc2c
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-06-12 14:56:55
# Regressor Bug: 1641737
# File Overlap Count: 2
# Description:
#   Bug 1641737 - P2. Use MozPromise with DocumentLoadListener and remove cycle. r=mattwoodrow.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78179
# ==============================================================================

diff -r 02f45f660107 -r 81ddcb21f057 netwerk/ipc/DocumentChannelParent.h
--- a/netwerk/ipc/DocumentChannelParent.h	Fri Jun 12 05:56:26 2020 +0000
+++ b/netwerk/ipc/DocumentChannelParent.h	Fri Jun 12 05:56:28 2020 +0000
@@ -7,8 +7,8 @@
 #ifndef mozilla_net_DocumentChannelParent_h
 #define mozilla_net_DocumentChannelParent_h
 
+#include "mozilla/net/DocumentLoadListener.h"
 #include "mozilla/net/PDocumentChannelParent.h"
-#include "mozilla/net/DocumentLoadListener.h"
 
 namespace mozilla {
 namespace dom {
@@ -17,12 +17,10 @@
 namespace net {
 
 /**
- * An implementation of ADocumentChannelBridge that forwards all changes across
- * to DocumentChannelChild, the nsIChannel implementation owned by a content
- * process docshell.
+ * An actor that forwards all changes across to DocumentChannelChild, the
+ * nsIChannel implementation owned by a content process docshell.
  */
-class DocumentChannelParent final : public ADocumentChannelBridge,
-                                    public PDocumentChannelParent {
+class DocumentChannelParent final : public PDocumentChannelParent {
  public:
   NS_INLINE_DECL_REFCOUNTING(DocumentChannelParent, override);
 
@@ -33,45 +31,27 @@
 
   // PDocumentChannelParent
   bool RecvCancel(const nsresult& aStatus) {
-    if (mParent) {
-      mParent->Cancel(aStatus);
+    if (mDocumentLoadListener) {
+      mDocumentLoadListener->Cancel(aStatus);
     }
     return true;
   }
   void ActorDestroy(ActorDestroyReason aWhy) override {
-    if (mParent) {
-      mParent->DocumentChannelBridgeDisconnected();
-      mParent = nullptr;
+    if (mDocumentLoadListener) {
+      mDocumentLoadListener->Cancel(NS_BINDING_ABORTED);
     }
   }
 
  private:
-  // DocumentChannelListener
-  void DisconnectChildListeners(nsresult aStatus,
-                                nsresult aLoadGroupStatus) override {
-    if (CanSend()) {
-      Unused << SendDisconnectChildListeners(aStatus, aLoadGroupStatus);
-    }
-    mParent = nullptr;
-  }
-
-  void Delete() override {
-    if (CanSend()) {
-      Unused << SendDeleteSelf();
-    }
-  }
-
-  ProcessId OtherPid() const override { return IProtocol::OtherPid(); }
-
   RefPtr<PDocumentChannelParent::RedirectToRealChannelPromise>
   RedirectToRealChannel(
       nsTArray<ipc::Endpoint<extensions::PStreamFilterParent>>&&
           aStreamFilterEndpoints,
-      uint32_t aRedirectFlags, uint32_t aLoadFlags) override;
+      uint32_t aRedirectFlags, uint32_t aLoadFlags);
 
   virtual ~DocumentChannelParent();
 
-  RefPtr<DocumentLoadListener> mParent;
+  RefPtr<DocumentLoadListener> mDocumentLoadListener;
 };
 
 }  // namespace net